WS2012R2/lisa/setupscripts/install_kernel_rpm.ps1

```diff
@@ -0,0 +1,145 @@
+#>
+
+param([string] $vmName, [string] $hvServer, [string] $testParams)
+
+#
+# Main script
+#
+# Check input arguments
+if ($vmName -eq $null) {
+    "Error: VM name is null"
+    return $retVal
+}
+
+if ($hvServer -eq $null) {
+    "Error: hvServer is null"
+    return $retVal
+}
+
+$params = $testParams.Split(";")
+foreach ($p in $params) {
+    $fields = $p.Split("=")
+
+    switch ($fields[0].Trim()) {
+        "rootDir"   { $rootDir = $fields[1].Trim() }
+        "sshKey" { $sshKey  = $fields[1].Trim() }
+        "ipv4"   {$ipv4 = $fields[1].Trim()}
+        "distro"   {$distro = $fields[1].Trim()}
+		"localPath"   {$localPath = $fields[1].Trim()}
+        default  {}
+    }
+}
+
+if ($null -eq $sshKey) {
+    "Error: Test parameter sshKey was not specified"
+    return $False
+}
+
+if ($null -eq $ipv4) {
+    "Error: Test parameter ipv4 was not specified"
+    return $False
+}
+
+if (-not $rootDir) {
+    "Warn : rootdir was not specified"
+}
+else {
+    cd $rootDir
+}
+
+$summaryLog = "${vmName}_summary.log"
+del $summaryLog -ErrorAction SilentlyContinue
+
+# Source TCUtils.ps1 for getipv4 and other functions
+if (Test-Path ".\setupScripts\TCUtils.ps1")
+{
+    . .\setupScripts\TCUtils.ps1
+}
+else
+{
+    "ERROR: Could not find setupScripts\TCUtils.ps1"
+    return $false
+}
+$kernel = "test-artifacts"
+# Send files to VM
+if ($distro -eq "rhel" -or $distro -eq "centos") {
+    $fileExtension = "rpm"
+}
+if ($distro -eq "ubuntu") {
+    $fileExtension = "deb"
+}
+if (Test-Path $localPath\*.$fileExtension)
+{
+    $files = Get-ChildItem $localPath -Filter *.${fileExtension}
+}
+else {
+    Write-Output "Error: $fileExtension files are not present! $test" | Tee-Object -Append -file $summaryLog
+    return $false
+}
+
+SendCommandToVM $ipv4 $sshKey "mkdir /tmp/$kernel/"
+foreach ($file in $files){
+    $filePath = $file.FullName
+
+    # Copy file to VM
+    SendFileToVM $ipv4 $sshKey $filePath "/tmp/$kernel/"
+
+    Start-Sleep -s 1
+}
+Write-Output "All files have been sent to VM. Will proceed with installing the new kernel" | Tee-Object -Append -file $summaryLog
+
+if ($distro -eq "rhel" -or $distro -or $distro -eq "centos") {
+    # Install RPMs
+    SendCommandToVM $ipv4 $sshKey "yum localinstall -y /tmp/$kernel/*.rpm"
+    Start-Sleep -s 100
+
+    # Update daemon startup paths
+    SendCommandToVM $ipv4 $sshKey "sed -i 's,ExecStart=/usr/sbin/hypervkvpd,ExecStart=/usr/sbin/hypervkvpd -n,' /usr/lib/systemd/system/hypervkvpd.service"
+    SendCommandToVM $ipv4 $sshKey "sed -i 's,ExecStart=/usr/sbin/hypervvssd,ExecStart=/usr/sbin/hypervvssd -n,' /usr/lib/systemd/system/hypervvssd.service"
+    SendCommandToVM $ipv4 $sshKey "sed -i 's,ExecStart=/usr/sbin/hypervfcopyd,ExecStart=/usr/sbin/hypervfcopyd -n,' /usr/lib/systemd/system/hypervfcopyd.service"
+
+    # Modify GRUB2
+    SendCommandToVM $ipv4 $sshKey "grub2-mkconfig -o /boot/grub2/grub.cfg ; grub2-set-default 0"
+
+    # Modify GRUB
+    Start-Sleep -s 200
```

Comment:
if the sendcommand to the vm is async, how do you make sure the command is okay?